packages/opentelemetry-exporter-zipkin/src/utils.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { GetHeaders } from './types';
+
+export function prepareGetHeaders(
```

Comment:
Basically Zipkin has 3 transport layers, node with http and web with xml-xthp-request and sendBeacon. If it was only one it would be easier.
For web if user defines a header, then sendBeacon cannot be used but only XHR. To make an improvement and not to check the config if headers exists in constructor a new function is being created with either headers or without them. This covers majority of cases so instead of parsing the same headers with every request the new function was created that did that once only. Now despite "entry" headers we have also interceptors. This is when fun begins. If user doesn't create any interceptor all remains as it was, if it does the only way is to create a new function with new headers from interceptor. And lastly you have node and web and to distinguish between those 2 and to be able not to check for headers every time if  nothing was configured a prepare function is called from the correct platform.

Reply:
Hmm ok seems fine. Just a little odd to see it but the explanation seems reasonable.